api/v1alpha1/vineyardruntime_types.go

```diff
@@ -0,0 +1,220 @@
+	// For Worker, support the following options.
+	//
+	//   vineyardd.reserve.memory: (Bool) where to reserve memory for vineyardd
+	//                             If set to true, the memory quota will be counted to the vineyardd rather than the application.
+	//   vineyardd.external.etcd.endpoint: (String) the endpoint of external etcd cluster
+	//
+	//   E,g. {"vineyardd.external.etcd.endpoint": "etcd-svc.etcd-namespace.svc.cluster.local:2379" or "10.244.3.4:2379"}
+	//
+	//   Default value is as follows.
+	//
+	//     vineyardd.reserve.memory: "true"
+	//
+	// +optional
+	Options map[string]string `json:"options,omitempty"`
+
+	// Resources contains the resource requirements and limits for the Vineyard component.
+	// Default is not set.
+	// For Worker, when the options contains vineyardd.reserve.memory=true,
+	// the resources.request.memory for worker should be greater than tieredstore.levels[0].quota(aka vineyardd shared memory)
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// VolumeMounts specifies the volumes listed in ".spec.volumes" to mount into the vineyard runtime component's filesystem.
+	// It is useful for specifying a persistent storage.
+	// Default is not set.
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+}
+
+// MasterSpec defines the configurations for Vineyard Master component
+// which is also regarded as the Etcd component in Vineyard.
+// For more info about Vineyard, refer to <a href="https://v6d.io/">Vineyard</a>
+type MasterSpec struct {
+	// The component configurations for Vineyard Master
+	// +optional
+	VineyardCompTemplateSpec `json:",inline"`
+
+	// Whether to deploy the Vineyard Master component
+	// Default is true
+	// If set to false, the Etcd component will not be deployed,
+	// which means the Vineyard Worker component will use an external Etcd cluster using http protocol.
+	// In this case, the worker's options should be set as follows.
+	// worker:
+	//   options:
+	//     "vineyardd.external.etcd.endpoint": "etcd-svc.etcd-namespace.svc.cluster.local:2379" or "10.244.3.4:2379"
+	//
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+// VineyardFuseSpec holds the configurations for vineyard fuse
+type VineyardFuseSpec struct {
+	// Image for Vineyard Fuse
+	// Default is `vineyardcloudnative/vineyard-mount-socket`
+	// +optional
+	Image string `json:"image,omitempty"`
+
+	// Image Tag for Vineyard Fuse
+	// Default is `latest`
+	// +optional
+	ImageTag string `json:"imageTag,omitempty"`
+
+	// Image pull policy for Vineyard Fuse
+	// Default is `IfNotPresent`
+	// Available values are `Always`, `IfNotPresent`, `Never`
+	// +optional
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
+
+	// Resources contains the resource requirements and limits for the Vineyard Fuse.
+	// Default is not set.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+}
```

Comment:
Does Vineyard Fuse need `NodeSelector` and `CleanPolicy` field like other runtimes? These fields are used at  `GetRuntimeInfo` method in the `runtime.go` file

Reply:
I think the answer is no, as the vineyard fuse only mounts the vineyard socket from hostpath to the application pod's PVC, so it must be co-located with application Pod. Adding a new `nodeselector` may mix the deployment of Vineyard Fuse. As for `CleanPolicy`, the only strategy is `OnDemand`. When the application pod is deleted, the fuse pod will be deleted after that. What do you think? @xliuqq

Reply:
@dashanji As far as I know，fuse is always co-located with application pod because the CSI plugin sets the node label `fluid.io/f-{namespace}-{name}` (fuse daemonset node selector expression), the `nodeselector`  defines their(fuse/app) placement requirements.
The [proposal](https://github.com/fluid-cloudnative/fluid/issues/3528) says it supports RPC between app pod and vineyard pod, so we can use `nodeselector` to place the fuse/app pod with requirements ?
As for `CleanPolicy`, do you mean vineyard fuse not support `OnRuntimeDeletedCleanPolicy` ?

Reply:
> The https://github.com/fluid-cloudnative/fluid/issues/3528 says it supports RPC between app pod and vineyard pod. So we can use nodeselector to place the fuse/app pod with requirements?

I think the original nodeselector `fluid.io/f-{namespace}-{name}` is enough. If the app pod uses the RPC, we don't need the vineyard fuse (which mainly mounts vineyard socket from the hostpath) at all.

> As for CleanPolicy, do you mean vineyard fuse not support OnRuntimeDeletedCleanPolicy?

Sorry for the mistake. I have checked the `NodeStageVolume`. Both `OnDemand` and `OnRuntimeDeleted` are supported, and the difference is whether the fuse pod still exists after the app pod is deleted. Maybe `OnRuntimeDeletedCleanPolicy` is more better.

Reply:
@xliuqq Thanks for the advice. I have added the `OnRuntimeDeletedCleanPolicy`. 
Maybe we could add `nodeselector` after the users make a request.